### DISCUSSION TITLE
Click on Pane clears the selected nodes and edges. Is there a way to prevent it?

### DISCUSSION THREAD

**Discussion by ratnajeetshyamkunwar** [xyflow/xyflow#2769] [Q&A] [answered]
### Describe the Bug

When clicking on the pane it clears the selection. Selected nodes and edges get deselected. Is there a way to prevent it?

### Steps to Reproduce the Bug or Issue

1. select one or multiple nodes
2. click on the pane
3. Selection gets cleared

### Expected behavior

If I selected multiple Items and by mistake, I clicked on the pane, the selection gets cleared. I should be able to prevent accidental deselection.

### Platform

- OS: macOS
- Browser: Chrome
- Version: 11.4.0

**Comment by ratnajeetshyamkunwar** [comment, marked as answer]
currently, I have achieved this using this trick. But it might not be full proof.

`onPaneClick={e => {
              setTimeout(() => {
                setNodes(nodes);
              }, 0);
          }}`

**Reply by zhenghan2017** [reply]
Hi，Bro.
Is there any other way? if use the solution will refresh component and UI flash across when you have custom elements.

**Reply by ratnajeetshyamkunwar** [reply]
@zhenghan2017 I haven't worked on it for a long time. I did not try to find any other solution. Slight flickering was not too big of an issue for me.

**Reply by thibaultbrunet** [reply]
Hi @zhenghan2017 

I ran into the same issue, I've put a custom selected field in the data for my nodes. which is not impacted by the default library behaviour.
